contracts/apps/disputable/DisputableApp.sol

```diff
@@ -0,0 +1,186 @@
+/*
+ * SPDX-License-Identitifer:    GPL-3.0-or-later
+ */
+
+pragma solidity 0.4.24;
+
+import "./IAgreement.sol";
+import "./IDisputable.sol";
+import "../AragonApp.sol";
+import "../../lib/token/ERC20.sol";
+import "../../lib/math/SafeMath64.sol";
+
+
+contract DisputableApp is IDisputable, AragonApp {
+    /* Validation errors */
+    string internal constant ERROR_AGREEMENT_NOT_SET = "DISPUTABLE_AGREEMENT_NOT_SET";
+    string internal constant ERROR_AGREEMENT_ALREADY_SET = "DISPUTABLE_AGREEMENT_ALREADY_SET";
+    string internal constant ERROR_SENDER_NOT_AGREEMENT = "DISPUTABLE_SENDER_NOT_AGREEMENT";
+
+    // This role is not required to be validated in the Disputable app, the Agreement app is already doing the check
+    // bytes32 public constant CHALLENGE_ROLE = keccak256("CHALLENGE_ROLE");
+    bytes32 public constant CHALLENGE_ROLE = 0xef025787d7cd1a96d9014b8dc7b44899b8c1350859fb9e1e05f5a546dd65158d;
+
+    // bytes32 public constant SET_AGREEMENT_ROLE = keccak256("SET_AGREEMENT_ROLE");
+    bytes32 public constant SET_AGREEMENT_ROLE = 0x8dad640ab1b088990c972676ada708447affc660890ec9fc9a5483241c49f036;
+
+    // bytes32 internal constant AGREEMENT_POSITION = keccak256("aragonOS.appStorage.agreement");
+    bytes32 internal constant AGREEMENT_POSITION = 0x6dbe80ccdeafbf5f3fff5738b224414f85e9370da36f61bf21c65159df7409e9;
+
+    event AgreementSet(IAgreement indexed agreement);
+
+    modifier onlyAgreement() {
+        require(address(_getAgreement()) == msg.sender, ERROR_SENDER_NOT_AGREEMENT);
+        _;
+    }
+
+    /**
+    * @notice Set disputable agreements to `_agreement`
+    * @param _agreement Agreement instance to be linked
+    */
+    function setAgreement(IAgreement _agreement) external auth(SET_AGREEMENT_ROLE) {
+        IAgreement agreement = _getAgreement();
+        require(agreement == IAgreement(0) || _agreement == IAgreement(0), ERROR_AGREEMENT_ALREADY_SET);
```

Comment:
Wondering if we should expose the ability to unset the Agreement.

We were discussing before that it'd be a pretty tricky migration, and perhaps just won't let apps do it?

Reply:
Note that this is actually allowing to unset the agreement, it only requires the current agreement is not set in that case

Reply:
Not sure I understand this; doesn't the `_agreement == IAgreement(0)` allow us to unset it?

If anything, it seems like we might want `_agreement != IAgreement(0)`?

Reply:
> Not sure I understand this; doesn't the `_agreement == IAgreement(0)` allow us to unset it?

Yes, but I think the idea was to provide the ability to unset disputable apps from Agreement instances, isn't that the case?